src/javaObject.cpp

```diff
@@ -54,6 +67,18 @@
       v8::Handle<v8::String> methodNameSync = NanNew<v8::String>((methodNameStr + "Sync").c_str());
       v8::Local<v8::FunctionTemplate> methodCallSyncTemplate = NanNew<v8::FunctionTemplate>(methodCallSync, methodName);
       funcTemplate->PrototypeTemplate()->Set(methodNameSync, methodCallSyncTemplate->GetFunction());
+
+      if (promisifying) {
+        v8::Local<v8::Object> recv = asyncOptions; // just a dummy receiver
+        v8::Local<v8::Value> argv[] = { methodCallTemplate->GetFunction() };
+        v8::Local<v8::Value> result = promisify->Call(recv, 1, argv);
+        if (!result->IsFunction()) {
+          printf("Promisified result is not a function.\n");
```

Comment:
We may want to fail hard here. I'm pretty sure everything after this will not like it if result is not a function. In `checkJavaException` node-java `assert(false)` after printing the error. It would be nice to throw an exception but I'm not sure if that's allowed in the New method.